File: ITing-backend/src/main/java/com/iting/jobportal/admin/service/AdminReportService.java
package com.iting.jobportal.admin.service;

import com.iting.jobportal.admin.dto.response.ReportStatsResponse;
import com.iting.jobportal.admin.entity.UserReport;
import com.iting.jobportal.user.dto.request.ReportRequest;
import org.springframework.data.domain.Page;

public interface AdminReportService {

    Page<UserReport> getReports(String status, String type, String targetType, String priority, String search, int page, int size);

    UserReport createReport(Long reporterId, ReportRequest request);

    UserReport handleReport(Long adminId, Long reportId, String status, String note);

    ReportStatsResponse getReportStats();

    UserReport getReportById(Long id);

}

File: ITing-frontend/src/services/adminReportService.js
import axiosInstance from "../utils/axiosInstance";

const adminReportService = {
  getReports: async (params) => {
    try {
      const response = await axiosInstance.get("/admin/reports", { params });
      return response;
    } catch (error) {
      console.error("[API] GET /admin/reports - error:", error);
      throw error;
    }
  },

  getReportStats: async () => {
    try {
      const response = await axiosInstance.get("/admin/reports/stats");
      return response;
    } catch (error) {
      console.error("[API] GET /admin/reports/stats - error:", error);
      throw error;
    }
  },

  handleReport: async (id, status, note) => {
    try {
      const response = await axiosInstance.put(`/admin/reports/${id}/handle`, null, {
        params: { status, note },
      });
      return response;
    } catch (error) {
      console.error(`[API] PUT /admin/reports/${id}/handle - error:`, error);
      throw error;
    }
  },

  getReportById: async (id) => {
    try {
      const response = await axiosInstance.get(`/admin/reports/${id}`);
      return response;
    } catch (error) {
      console.error(`[API] GET /admin/reports/${id} - error:`, error);
      throw error;
    }
  },
};

export default adminReportService;
